### DISCUSSION TITLE
Windows version rebuild-scratch-org

### DISCUSSION THREAD

**Discussion by PeterLin888** [wimvelzeboer/fflib-apex-extensions#8] [Q&A]
Hello William,

Do you have a Windows version of rebuild-scratch-org? It seemed to me that the version below was a Linux or macOS version.
https://github.com/wimvelzeboer/fflib-apex-extensions/blob/main/dev-tools/rebuild-scratch-org

Thanks,
Peter

**Comment by wimvelzeboer** [comment]
@PeterLinBiz The rebuild-scratch-org script is indeed based on Linux/MacOs. The script is taken from the [Falcon Template](https://github.com/sfdx-isv/sfdx-falcon-template) which Salesforce made available for ISVs. Initially it only contained a Linux based version, but I just had a look at it, and it seems that they have extended it to Windows. 

The folder dev-tools is Linux based, and dev-tools-win is Windows based.
I don't have a Windows machine, so I am not really able to validate anything. 
Would you be able to copy paste that and test to see if you can get it running for Windows?

The only bit I have modified in its original is that I added the following just before it does a "pushMetadata"
```
mkdir $PROJECT_ROOT/temp
cd $PROJECT_ROOT/temp
rm -Rf fflib-apex-mocks/
rm -Rf fflib-apex-common/
git clone https://github.com/apex-enterprise-patterns/fflib-apex-mocks.git
git clone https://github.com/apex-enterprise-patterns/fflib-apex-common.git

sfdx force:source:deploy -p ./fflib-apex-mocks/sfdx-source -u $SCRATCH_ORG_ALIAS
sfdx force:source:deploy -p ./fflib-apex-common/sfdx-source -u $SCRATCH_ORG_ALIAS
```

It does a checkout of the mocks and commons (in a temp folder outside the project directories), and deploys those folders.
